Libplanet.Explorer/Queries/BlockQuery.cs

```diff
@@ -43,13 +43,30 @@ public BlockQuery()
             Field<BlockType<T>>(
                 "block",
                 arguments: new QueryArguments(
-                    new QueryArgument<IdGraphType> { Name = "hash" }
+                    new QueryArgument<IdGraphType> { Name = "hash" },
+                    new QueryArgument<IdGraphType> { Name = "index" }
                 ),
                 resolve: context =>
                 {
-                    HashDigest<SHA256> hash = HashDigest<SHA256>.FromString(
-                        context.GetArgument<string>("hash"));
-                    return Query<T>.GetBlock(hash);
+                    string hash = context.GetArgument<string>("hash");
+                    long? index = context.GetArgument<long?>("index", null);
+
+                    if (!(hash is null ^ index is null))
```

Comment:
Your intention is to prevent a user giving two arguments *at a time*, right?  How about adding a conditional `if (!(index is null))` inside the below `if (hash is string hashNotNull)` block instead?

```csharp
if (hash is string hashNotNull)
{
    if (!(index is null))
    {
        throw new GraphQL.ExecutionError(
            "The parameters hash and index are mutually exclusive; " +
            "give only one at a time.");
    }

    return Query<T>.GetBlockByHash(HashDigest<SHA256>.FromString(hashNotNull));
}
```